src/gui/editorwidgets/qgstexteditwrapper.cpp

```diff
@@ -64,7 +64,8 @@ QVariant QgsTextEditWrapper::value() const
   }
 
   QVariant res( v );
-  if ( field().convertCompatible( res ) )
+  // treat VariantMap fields including JSON differently
+  if ( field().convertCompatible( res ) && field().type() != QVariant::Map )
```

Comment:
Switch those conditions, so the conversion is only done if it's used after

```suggestion
  if ( field().type() != QVariant::Map && field().convertCompatible( res ) )
```